In [ ]:
import pandas as pd

df = pd.read_csv('result.csv')
df.head()

In [ ]:
df = df.filter(['id', 'text'])
df = df[df['text'].map(lambda x: len(x.split()) >= 3)]
df.head()

id                                               text
0  1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1  1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2  1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3  1389951640610902020                                     tbtb gelisah 🤕
4  1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...

In [ ]:
len(df)

2751

In [ ]:
df = df.head(1000)
len(df)

1000

In [ ]:
df

id                                               text
0     1389953924879110144  abis minum kopi perasaan mesti gelisah gatau k...
1     1389952303407329283          sumpah ya dari pagi rasanya gelisah terus
2     1389952181676122116  Catia hari sabtu.Presntasi Statin hari sabtu.....
3     1389951640610902020                                     tbtb gelisah 🤕
4     1389950790266023936  Memasuki jam jam rawan... Seliweran mulu di ga...
...                   ...                                                ...
1019  1389076387684700162  Moots kalau berusaha tidur hadap kanan, hadap ...
1020  1389075326664286209  aku diem aja mbak nya cemas gmn kl aku gerak d...
1021  1389074928893267971  Tenang saja, semua akan baik baik saja. Tidak ...
1022  1389074441762611201  Tak sukanya orang beli baju berebut ni lepastu...
1023  1389074401883086850  Kalau pagi2 resah, gelisah, anxious itu tanda ...

[1000 rows x 2 columns]

Three annotators, so each of them will get 300 tweets + 100 shared tweets. In total:

$$3\times 300 + 100 = 1000$$

In [ ]:
split1, split2, split3, share = df[0:300], df[300:600], df[600:900], df[900:]

In [ ]:
split1.to_csv('split1.csv', index=False)
split2.to_csv('split2.csv', index=False)
split3.to_csv('split3.csv', index=False)
share.to_csv('share.csv', index=False)

# Creating DATD dataset

Merging back responses

In [13]:
split1 = pd.read_csv('split1.csv')
split2 = pd.read_csv('split2.csv')
split3 = pd.read_csv('split3.csv')
share = pd.read_csv('share.csv')

In [9]:
def load_response(response_name):
  res = []
  with open('{}.response'.format(response_name)) as f:
    res = [int(r.strip()) for r in f.readlines()]
  return res

In [3]:
rsplit1, rsplit2, rsplit3 = load_response('split1'), load_response('split2'), load_response('split3')

In [4]:
rshare1, rshare2, rshare3 = load_response('share-1'), load_response('share-2'), load_response('share-3')

In [5]:
print(rshare1)
print(rshare2)
print(rshare3)

[0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]
[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0]
[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]


In [6]:
rshare = []

# three annotators, so tie is not possible
for i in range(len(rshare1)):
  zero = 0
  one = 0
  if rshare1[i] == 0:
    zero += 1
  else:
    one += 1
  
  if rshare2[i] == 0:
    zero += 1
  else:
    one += 1
  
  if rshare3[i] == 0:
    zero += 1
  else:
    one += 1
  
  if zero > one:
    rshare.append(0)
  else:
    rshare.append(1)

print(rshare)

[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0]


In [10]:
annotated_share = share.assign(label=rshare)

In [14]:
annotated_split1 = split1.assign(label=rsplit1)
annotated_split2 = split2.assign(label=rsplit2)
annotated_split3 = split3.assign(label=rsplit3)

In [15]:
annotated_dataset = pd.concat([annotated_split1, annotated_split2, annotated_split3], axis=0)

In [16]:
annotated_dataset

id  ... label
0    1389953924879110144  ...     0
1    1389952303407329283  ...     1
2    1389952181676122116  ...     1
3    1389951640610902020  ...     0
4    1389950790266023936  ...     0
..                   ...  ...   ...
295  1389172826595160067  ...     0
296  1389169768003895298  ...     1
297  1389168219542286342  ...     0
298  1389168028416233472  ...     0
299  1389167572763901952  ...     0

[900 rows x 3 columns]

In [17]:
from collections import Counter

label_counter = Counter(annotated_dataset['label'])
print(label_counter)

label_counter = Counter(annotated_share['label'])
print(label_counter)

Counter({0: 523, 1: 377})
Counter({0: 68, 1: 32})


Dropping `id` because it's not relevant for classification

In [18]:
annotated_dataset = annotated_dataset.drop('id', axis=1)
annotated_share = annotated_share.drop('id', axis=1)

In [19]:
print(annotated_dataset)

                                                  text  label
0    abis minum kopi perasaan mesti gelisah gatau k...      0
1            sumpah ya dari pagi rasanya gelisah terus      1
2    Catia hari sabtu.Presntasi Statin hari sabtu.....      1
3                                       tbtb gelisah 🤕      0
4    Memasuki jam jam rawan... Seliweran mulu di ga...      0
..                                                 ...    ...
295  Habis muncul bimbingan setelah menghilang berb...      0
296  Resah dan gelisah deh mimin, dah sebulan lewat...      1
297  Jangan membandingkan dirimu dengan orang lain....      0
298  Senin, 03 Mei 2020 / 21 Ramadhan 1442.Hari bai...      0
299       Gelisah, galau kalau seharian ga ada kabar..      0

[900 rows x 2 columns]


In [20]:
print(annotated_share)

                                                 text  label
0   Aku merindukan tempatku yang tenang, tanpa ras...      1
1   Gelisah tanpa sebab. Terasa seperti punya kein...      1
2   Sekarang kita masih sama, suka bingung. Suka c...      0
3   aku mula lah gelisah bila dia outstation , kal...      0
4   Kasihan kamu, hidup dalam tekanan dan selalu d...      0
..                                                ...    ...
95  Moots kalau berusaha tidur hadap kanan, hadap ...      0
96  aku diem aja mbak nya cemas gmn kl aku gerak d...      0
97  Tenang saja, semua akan baik baik saja. Tidak ...      0
98  Tak sukanya orang beli baju berebut ni lepastu...      0
99  Kalau pagi2 resah, gelisah, anxious itu tanda ...      0

[100 rows x 2 columns]


In [21]:
annotated_dataset.to_csv('datd_train.csv', index=False)
annotated_share.to_csv('datd_test.csv', index=False)

# Creating DATD+Rand dataset

In [22]:
annotated_dataset[annotated_dataset['label'] == 1]

text  label
1            sumpah ya dari pagi rasanya gelisah terus      1
2    Catia hari sabtu.Presntasi Statin hari sabtu.....      1
11   Pernah gasi ngerasain tiap hari gelisah karena...      1
13   Aku makin gelisah, dan aku menangis sejadi jad...      1
14       ko aku dekdekan kenapa ya??aku gelisah banget      1
..                                                 ...    ...
280  Udah ngalihin fokus ke yang lain tapi tetep cemas      1
285  Kenapa ya cemas susah ilang? Deg2an panik itu ...      1
290  Kalo mau jujur sih gelisah sejak hari itu wala...      1
291  tiap malem overthinking, cemas, gelisah..tiap ...      1
296  Resah dan gelisah deh mimin, dah sebulan lewat...      1

[377 rows x 2 columns]

Based on the paper, DATD+Rand contained $10.5\%$ positive instances, so we will solve

$$\dfrac{377}{377+x} = 0.105$$

and we get $x = \lfloor 3213.48 \rfloor \approx 3213$

In [ ]:
random_df = pd.read_csv('random_dataset.csv')
random_df = random_df.filter(['text'])
random_df['label'] = 0
random_df.head(3213)

text  label
0     Gr gr mimpiin mantan crush jdi gamon gue ya Allah      0
1                                  wkwkwk apaansi lawak      0
2     Cuti2 pun standby depan pc 😅 sebab office tak ...      0
3            Alhamdulillah meskipun hanya numpang lewat      0
4     konsep teaser pernikahan pake lagu day6 - only...      0
...                                                 ...    ...
3208                        JA JA JA F POR EL JYP AHHHH      0
3209       Budak sekarang pasal tudung pun nak sensitif      0
3210  markah test semua dah keluar, cam okay lah sem...      0
3211  normalise hating someone by yourself bukan per...      0
3212       dilengangnya malam menuju minggu kau dimana~      0

[3213 rows x 2 columns]

In [ ]:
annotated_random = pd.concat([annotated_dataset[annotated_dataset['label'] == 1], random_df.head(3213)], axis=0)
annotated_random

text  label
1             sumpah ya dari pagi rasanya gelisah terus      1
2     Catia hari sabtu.Presntasi Statin hari sabtu.....      1
11    Pernah gasi ngerasain tiap hari gelisah karena...      1
14    Aku makin gelisah, dan aku menangis sejadi jad...      1
15        ko aku dekdekan kenapa ya??aku gelisah banget      1
...                                                 ...    ...
3208                        JA JA JA F POR EL JYP AHHHH      0
3209       Budak sekarang pasal tudung pun nak sensitif      0
3210  markah test semua dah keluar, cam okay lah sem...      0
3211  normalise hating someone by yourself bukan per...      0
3212       dilengangnya malam menuju minggu kau dimana~      0

[3590 rows x 2 columns]

In [ ]:
annotated_random.to_csv('datd_rand.csv', index=False)